### PR TITLE
python3Packages.azure-mgmt-automation: init at 1.0.0

### PR DESCRIPTION
**Pull request by wfdewith**
This is one of the Azure SDK packages that hasn't been packaged in Nix yet. It is used by the [Azure Ansible collection](https://docs.ansible.com/ansible/latest/collections/azure/azcollection/index.html), which is why I needed it packaged here.

[Link](https://github.com/Azure/azure-sdk-for-python/tree/azure-mgmt-automation_1.0.0/sdk/automation/azure-mgmt-automation/)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/azure-mgmt-automation/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  fetchPypi,
+  msrest,
+  azure-common,
+  azure-mgmt-core,
+}:
+
+buildPythonPackage rec {
+  pname = "azure-mgmt-automation";
+  version = "1.0.0";
+  format = "setuptools";
```
Comment (fabaff) [inline pkgs/development/python-modules/azure-mgmt-automation/default.nix:13]:
```suggestion
  pyproject = true;
```

and add `build-system = [ setuptools ];`

File: pkgs/development/python-modules/azure-mgmt-automation/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    extension = "zip";
+    hash = "sha256-pJ0tQT/vVwEMs2Fh5bwFZgG418bQW9PLBaE1Eu8pHh4=";
+  };
+
+  propagatedBuildInputs = [
```
Comment (fabaff) [inline pkgs/development/python-modules/azure-mgmt-automation/default.nix:21]:
```suggestion
  dependencies = [
```

File: pkgs/development/python-modules/azure-mgmt-automation/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  ];
+
+  # has no tests
+  doCheck = false;
+
+  pythonImportsCheck = [ "azure.mgmt.automation" ];
+
+  meta = with lib; {
+    description = "This is the Microsoft Azure Automation Client Library";
+    homepage = "https://github.com/Azure/azure-sdk-for-python";
```
Comment (fabaff) [inline pkgs/development/python-modules/azure-mgmt-automation/default.nix:34]:
```suggestion
    homepage = "https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/automation/azure-mgmt-automation";
    changelog = "https://github.com/Azure/azure-sdk-for-python/blob/azure-mgmt-automation_${version}/sdk/automation/azure-mgmt-automation/CHANGELOG.md";
```

**Comment by wfdewith** [comment]
Thanks, I copied one of the other `azure-mgmt-*` packages as a starting point, but it seems that those are not entirely up to date with the current Python packaging practices.

**Comment by fabaff** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 352108`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.azure-mgmt-automation</li>
    <li>python311Packages.azure-mgmt-automation.dist</li>
    <li>python312Packages.azure-mgmt-automation</li>
    <li>python312Packages.azure-mgmt-automation.dist</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
Looks good to me.

Seems fine to skip `typing-extensions>=4.3.0; python_version<'3.8.0'`.

**Pull request merged**